Dropdown/src/Dropdown.jsx

```diff
@@ -185,16 +185,17 @@ class Dropdown extends Component {
         const index = props.options.findIndex(this.findMatchingItem, this);
         if(index) {
 
-            const value = this.getOption(index).value;
+            const value = this.getOption(index);
 
-            this.setState({
-                closestValue: value,
-                currentIndex: index
-            },() => {
-                this.setState({dropdownText: ""});
-                this.dropdownSearch.value = "";
-                setTimeout(() => this.scrollToSelectedItem(),100);
-            });
+            if(value) {
+                this.setState({
+                    closestValue: value,
```

Comment:
You changed the set value from `this.getOption(index).value` to `this.getOption(index)`. I don't think it was intentional?

Reply:
ti was generating a runtime error in case of typed search string was not found. In this case, -1 is returned as an index and getOption(index) returned undefined.